ui/app/adapters/pki/role.js

```diff
@@ -34,6 +34,13 @@ export default class PkiRoleAdapter extends ApplicationAdapter {
     });
   }
 
+  updateRecord(store, type, snapshot) {
+    const { name, backend } = snapshot.record;
```

Comment:
backend is added to the model when the record is created-[see here](https://github.com/hashicorp/vault/blob/main/ui/lib/pki/addon/routes/roles/create.js#L12)